.github/workflows/ci.yml

```diff
@@ -81,15 +81,16 @@ jobs:
       fail-fast: false
       matrix:
         os: [ubuntu-latest, macos-latest]
+        rust: [nightly, stable]
     steps:
     - uses: actions/checkout@v4
     - name: Install Rust
-      run: rustup update stable
+      run: rustup update ${{ matrix.rust }}
```

Comment:
Instead of adding `+nightly` to all cargo command calls, I recommended using ` && rustup default ${{ matrix.rust }}` here.